src/protobuf.zig

```diff
@@ -1043,25 +1149,57 @@ inline fn is_tag_known(comptime field_desc: FieldDescriptor, tag_to_check: Extra
     return false;
 }
 
+fn getRootType(T: type) type {
+    return if (isZigProtobufManagedStruct(T))
+        T.UnderlyingType
+    else
+        return T;
+}
+
+fn DecodingContext(T: type) type {
+    return struct {
+        result: T,
+
+        const Self = @This();
+        const rootType = getRootType(T);
+
+        pub fn init(allocator: Allocator) !Self {
+            return if (isZigProtobufManagedStruct(T))
+                Self{ .result = try T.init(allocator) }
+            else
+                Self{ .result = pb_init(T, allocator) };
+        }
+
+        pub fn getTarget(self: *Self) *rootType {
+            return if (isZigProtobufManagedStruct(T))
+                self.result.getPointer()
+            else
+                return &self.result;
+        }
+    };
+}
```

Comment:
I think we can convert this struct to two functions with comptime parameters. AFAIK the struct is not being passed and only holds a value which can 1) be returned in `init` and 2) be passed to getTarget